frontend/src/features/documents/DownloadDocumentButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { LinkButton } from 'components/common/buttons';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import { FileDownload } from 'models/api/DocumentManagement';
+import { FaDownload } from 'react-icons/fa';
+
+import { useDocumentProvider } from './documentDetail/hooks/useDocumentProvider';
+
+export interface IDownloadDocumentButtonProps {
+  mayanDocumentId: number;
+  mayanFileId?: number;
+}
+
+const DownloadDocumentButton: React.FunctionComponent<IDownloadDocumentButtonProps> = props => {
```

Comment:
In my mind this component is doing a lot for a button, it seems like a mix of a display component and a container.

Up to you, but I'd be tempted to just pass in functions to handle the click event and then have a parent container/component handle the click. you should be able to pass any loading/disabled state logic via props.

Reply:
I thought so too. The problem is that we need the same functionality multiple times, including the visual aspect. I believe it encapsulated as a widget works better. Could be renamed given is not only a button